ouroboros-consensus/src/Ouroboros/Consensus/Node.hs

```diff
@@ -309,6 +309,13 @@ forkBlockProduction IS{..} =
           Just proof -> do
             (prevPoint, prevNo) <- prevPointAndBlockNo currentSlot <$>
                                      ChainDB.getCurrentChain chainDB
+
+            -- In this circumstance, it is required that we call 'syncState'
+            -- before 'getTxs' within this 'STM' transaction since we need to
+            -- guarantee that the transactions returned from 'getTxs' are
+            -- valid. Refer to the 'getTxs' documentation for more information.
```

Comment:
valid with respect to the current ledger state of the chain db. They were valid at some point before.